examples/dsa_hisa/hisa.py

```
@@ -0,0 +1,240 @@
+    # ------------------------------------------------------------------
+    block_topk_eff = min(block_topk, block_k_score.shape[-1])
+    topk_block_indices = torch.topk(
+        block_k_score.bfloat16(),
+        k=block_topk_eff,
+        dim=-1,
+        sorted=False,
+    ).indices  # [M, block_topk_eff] int64
+
+    # ------------------------------------------------------------------
+    # Stage 2: fp8 fine-grained Q·K MQA over only the selected
+    # blocks' raw tokens (block_topk_eff blocks × k_block_size tokens
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)